### PR TITLE
[DEVDOCS-5831/PAPI-2091]: Carts v3 and Store information V2, add store-level metafield's open api spec (#197)

### PR DESCRIPTION
**Pull request by bc-tgomez**
# [PAPI-2091] [DEVDOCS-5831]

## What changed?
- Store metafield's open api spec is now available in Store Information V2

## Release notes draft

* Store metafields is now available for use. Add metafield data to a merchant's store using this set of APIs. Metafields can be created, updated, and deleted on stores.

## Anything else?

[PAPI-2091]:
https://bigcommercecloud.atlassian.net/browse/PAPI-2091?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ping @bc-tayvs  

---------

# [DEVDOCS-]

## What changed?

*  Add Store metafields is now available. 

## Release notes draft

* 

## Anything else?

ping {names}

[PAPI-2091]: https://bigcommercecloud.atlassian.net/browse/PAPI-2091?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PAPI-2091]: https://bigcommercecloud.atlassian.net/browse/PAPI-2091?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DEVDOCS-5831]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-5831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/carts.v3.yml [hunk 1]
```diff
@@ -926,7 +926,7 @@ paths:
                 example-1:
                   value:
                     status: 404
-                    title: There was no order found with ID 1010
+                    title: There was no card found with ID 1010
```
Comment (slsriehl) [inline reference/carts.v3.yml:929]: card or cart?

File: reference/store_information.v2.yml [hunk 2]
```diff
@@ -149,6 +376,425 @@ components:
...
+        status:
+          type: integer
+        title:
+          type: string
+        type:
+          type: string
+        instance:
+          type: string
+        errors:
+          type: object
```
[the remark this answers is not in the document]
Comment (slsriehl) [reply reference/store_information.v2.yml:408]:
```suggestion
        errors:
          type: object
          properties: {}
```

File: reference/store_information.v2.yml [hunk 3]
```diff
@@ -149,6 +376,425 @@
...
+        Response payload for the BigCommerce API.
+      properties:
+        data:
+          type: array
+          items:
+            $ref: '#/components/schemas/Metafield'
+        errors:
+          type: array
+          description: Empty for 200 responses.
+          example: [ ]
```
[the remark this answers is not in the document]
Comment (slsriehl) [reply reference/store_information.v2.yml:606]:
```suggestion
        errors:
          type: array
          items: []
          description: Empty for 200 responses.
          example: [ ]
```

File: reference/store_information.v2.yml [hunk 4]
```diff
@@ -98,9 +100,234 @@ paths:
...
+              schema:
+                $ref: '#/components/schemas/MetaFieldCollectionDeleteResponseSuccess'
+        '422':
+          description: |
+            Response object for metafields deletion with partial success.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MetaFieldCollectionResponsePartialSuccess_DELETE'
+  '/store/metafields/{metafieldId}':
```
Comment (6juara9) [inline reference/store_information.v2.yml:243]: There should be no page, limit, key, namespace, direction params for singular get metafield endpoint.

Accept header is not really required, by default it should be `application/json` :) 

image

Comment (6juara9) [reply reference/store_information.v2.yml:243]: For get endpoint it is impossible to get the 409 :) and I'm not sure even about 422, it is just a GET, there is no input from the API client, also we don't want to document 5XX errors 

image

File: reference/store_information.v2.yml [hunk 5]
```diff
@@ -98,9 +100,234 @@ paths:
...
+                $ref: '#/components/schemas/ErrorResponse'
+        '500':
+          description: Internal Server Error
+      parameters:
+        - $ref: '#/components/parameters/PageParam'
+        - $ref: '#/components/parameters/LimitParam'
+        - $ref: '#/components/parameters/MetafieldKeyParam'
+        - $ref: '#/components/parameters/MetafieldNamespaceParam'
+        - $ref: '#/components/parameters/DirectionParam'
+    put:
```
Comment (6juara9) [inline reference/store_information.v2.yml:280]: - `Content-Type` and `Accept` headers are optional 

image

Comment (6juara9) [reply reference/store_information.v2.yml:280]: - No pagination for singular endpoint
- 422 status is missing

image

File: reference/store_information.v2.yml [hunk 6]
```diff
@@ -98,9 +100,234 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NotFound'
+              example:
+                value:
+                  status: 404
+                  title: There was no metafield found with ID 1010
+                  type: 'https://developer.bigcommerce.com/api-docs/getting-started/api-status-codes'
+    delete:
```
Comment (6juara9) [inline reference/store_information.v2.yml:317]: - `Accept` header is not required
- It is possible to have `404` status

image

**Review by 6juara9 — COMMENTED** [review]
I reviewed only singular endpoints and can see some issues in each of them, do you maybe need the access to the API just to double check all possible responses 🙂 ?

File: reference/store_information.v2.yml [hunk 7]
```diff
@@ -98,9 +100,234 @@ paths:
...
+                $ref: '#/components/schemas/MetaFieldCollectionResponse_Batch_POST_PUT'
+        '422':
+          description: |
+            Response object for metafields creation with partial success.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MetaFieldCollectionResponsePartialSuccess_POST_PUT'
+        '500':
+          description: Internal Server Error
```
Comment (kskomro) [inline reference/store_information.v2.yml:213]: We decided to don't document 500 errors.

File: reference/carts.v3.yml [hunk 8]
```diff
@@ -1048,7 +1048,7 @@ paths:
                 example-1:
                   value:
                     status: 404
-                    title: The resource at that URL was not found.
+                    title: There was no card found with ID 1010
```
[the remark this answers is not in the document]
Comment (bc-traciporter) [reply reference/carts.v3.yml]: I removed 404. This was for a PUT endpoint.

File: reference/store_information.v2.yml [hunk 9]
```diff
@@ -98,9 +100,166 @@ paths:
                       graphql_storefront_api_enabled: true
                       shopper_consent_tracking_enabled: true
                       multi_storefront_enabled: true
-                      storefront_limits: 
+                      storefront_limits:
                         active: 3
                         total_including_inactive: 4
+  '/store/metafields':
+    get:
+      summary: Get All Store Metafields
```
[the remark this answers is not in the document]
Comment (bc-traciporter) [reply reference/store_information.v2.yml:108]: @6juara9 When I tried it, it didn't work. This is why I did not add it. Maybe I was doing something wrong.

[the remark this answers is not in the document]
Comment (bc-traciporter) [reply reference/store_information.v2.yml:108]: Screenshot 2024-10-15 at 8 25 12 AM

[the remark this answers is not in the document]
Comment (bc-traciporter) [reply reference/store_information.v2.yml:108]: I got the update to work, so I added that one.

**Review by bc-tayvs — APPROVED** [review]

**Review by 6juara9 — APPROVED** [review]

**Pull request merged**